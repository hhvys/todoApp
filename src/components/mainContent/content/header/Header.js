import {connect} from 'react-redux';
import HeaderView from './HeaderView';
import {changeSorting, toggleDropDown} from "../../../../actions/actionCreaters";

const mapStateToProps = (state) => {

	let activeTab = state
					.tabs
					.find(tab => tab.tabId === state.activeTab);
	return {
		tabName: activeTab.tabName,
		query: state.searchQuery
	}
};

const mapDispatchToProps = (dispatch) => {
	return {
		changeSort: (sortBy) => dispatch(changeSorting(sortBy)),
		onClick: () => dispatch(toggleDropDown())
	};
};

const Header = connect(
	mapStateToProps,
	mapDispatchToProps
)(HeaderView);

export default Header;