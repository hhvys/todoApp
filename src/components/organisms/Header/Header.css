.div__header {
    left: 280px;
    z-index: 50;
}

.div__header .dorpdown-icons {
    margin-right: 280px;
}

.div__header .header-content {
    font-size: 20px;
    color: #fff;
    font-weight: 500;
}