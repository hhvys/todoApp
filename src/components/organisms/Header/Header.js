import React from 'react';
import styles from './Header.mod.scss';
import DropDownIcon from "../../molecules/DropDownIcon/DropDownIcon";
import {SORT} from "../../atoms/Icons/constants";
import DropDown from "../DropDown/DropDown";

class Header extends React.Component {

	dropDownHandler = (e) => {
		e.preventDefault();
		e.stopPropagation();
		this.setState((state) => {
			!state.pushDown ?
				window.addEventListener('click', this.dropDownHandler) :
				window.removeEventListener('click', this.dropDownHandler);
			return {
				pushDown: !state.pushDown
			}
		});
	};

	constructor(props) {
		super(props);
		this.state = {
			pushDown: false
		}
	}

	render() {
		const {
			collapsed,
			className,
			headerContent,
			onDropDownButtonClick,
			onDropDownTabClick,
			...props
		} = this.props;
		return (
			<div>
				<div
					className={`${className ? className : ''} ${collapsed ? styles.collapsed : ''} ${styles.header} fixed full-width d-flex align-items-center justify-content-between`}
					{...props}>
					<div className={`${styles.headerContent} pl-4`}>
						{headerContent}
					</div>
					<div className={`${styles.dropdownIcons} d-flex align-items-center justify-content-between`}
							 onClick={onDropDownButtonClick}
					>
						<DropDownIcon
							onClick={this.dropDownHandler}
							iconSymbol={SORT}
							iconText={"Sort"}
						/>
					</div>
				</div>
				{
					this.state.pushDown &&
					(
						<DropDown
							onTabClick={onDropDownTabClick}
							className={styles.dropdown}
						/>
					)
				}

			</div>

		);
	}
}

export default Header;
