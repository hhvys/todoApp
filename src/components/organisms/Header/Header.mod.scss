@import "../../../constants.mod";
.header {
    left: $side-bar-width+px;
    z-index: 1;
}

.collapsed {
    left: $collpased-side-bar-width+px;
}

.collapsed .dropdownIcons{
    margin-right: $collpased-side-bar-width+px;
}

.dropdownIcons {
    margin-right: $side-bar-width+px;
}

.headerContent {
    font-size: 20px;
    color: #fff;
    font-weight: 500;
}

.dropdown {
    width: 246px;
    position: fixed;
    right: 0;
    top: 44px;
    background-color: #678865;
    z-index: 50;
}
