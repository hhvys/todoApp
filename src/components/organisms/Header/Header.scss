.div__header {
    left: 280px;
    z-index: 50;
}

.div__header.collapsed {
    left: 42px;
}

.div__header.collapsed .dropdown-icons{
    margin-right: 42px;
}

.div__header .dropdown-icons {
    margin-right: 280px;
}

.div__header .header-content {
    font-size: 20px;
    color: #fff;
    font-weight: 500;
}