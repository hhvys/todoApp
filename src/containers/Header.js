import {connect} from 'react-redux';
import HeaderView from '../components/organisms/Header/Header';
import {changeSorting} from "../actions/actionCreaters";
import {getTabs} from "../reducers/tabs/tabs";
import {getSearchQuery} from "../reducers/searchQuery";
import {getActiveTab} from "../reducers/activeTab";
import {getCollapsedSideBar} from "../reducers/collapsedSideBar";


const mapStateToProps = (state) => {
	const tabs = getTabs(state);
	const searchQuery = getSearchQuery(state);
	const activeTab = getActiveTab(state);
	return ({
		headerContent: searchQuery ?
			searchQuery :
			tabs
				.find(tab => tab.tabId === activeTab)
				.tabName,
		collapsed: getCollapsedSideBar(state)
	})
};

const mapDispatchToProps = (dispatch) => {

	return ({
		onDropDownTabClick: (sortBy) => dispatch(changeSorting(sortBy)),
	})
};


const Header = connect(
	mapStateToProps,
	mapDispatchToProps
)(HeaderView);

export default Header;
