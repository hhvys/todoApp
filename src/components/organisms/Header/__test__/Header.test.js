import React from 'react';
import Header from '../Header';

describe('Header', () => {

	let props;
	let mountedHeader;

	const header = () => {
		if (!mountedHeader) {
			mountedHeader = shallow(
				<Header {...props}/>
			);
		}
		return mountedHeader;
	};

	beforeEach(() => {
		props = {
			collapsed: undefined,
			className: "ClassName",
			headerContent: undefined,
			onDropDownButtonClick: undefined,
			onDropDownTabClick: undefined,
		};
		mountedHeader = undefined;
	});

	it('should render', () => {
		const wrapperDiv = header().find('div').first();
		expect(toJSON(wrapperDiv)).toMatchSnapshot();
	});

	it('should have collapsed class when collapsed is true', () => {
		props.collapsed = true;
		const contentDiv = header().find('div').first().childAt(0);
		expect(contentDiv.props().className).toMatch('collapsed');
	});

	it('should not have collapsed when collapsed is not true', () => {
		props.collapsed = false;
		const contentDiv = header().find('div').first().childAt(0);
		expect(contentDiv.props().className).not.toMatch('collapsed');
	});

	it('should pass headerContent to contentDiv', () => {
		props.headerContent = "headerContent";
		const contentDiv = header().find('div').first().childAt(0);
		const div = contentDiv.childAt(0);
		expect(contentDiv.childAt(0).props().children).toBe(props.headerContent);
	});

});
