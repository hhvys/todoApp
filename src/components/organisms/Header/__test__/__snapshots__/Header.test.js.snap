// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div>
  <div
    className="ClassName  header fixed full-width d-flex align-items-center justify-content-between"
  >
    <div
      className="headerContent pl-4"
    />
    <div
      className="dropdownIcons d-flex align-items-center justify-content-between"
    >
      <DropDownIcon
        iconSymbol="SORT"
        iconText="Sort"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
